### PR TITLE
feat: add beta release workflow

### PR DESCRIPTION
**Pull request by danielbarion**
With the beta release workflow, all commits into the PR will trigger the beta release.

* the beta release always release a new beta version, following this format: 
`react-tooltip@{__VERSION__}-beta.{__PR_NUMBER__}.rc.{__BETA_VERSION__}` -> `react-tooltip@v0.0.11-beta.1.rc.1`
* to skip the workflow we need to add the following words to the commit message:
```
[skip ci]
[ci skip]
[no ci]
[skip actions]
[actions skip]
```

https://docs.github.com/en/actions/managing-workflow-runs/skipping-workflow-runs

image

image

https://www.npmjs.com/package/auto-beta-release?activeTab=versions

## Summary by CodeRabbit

- **New Features**
	- Introduced automated workflows for beta releases, enhancing the release process.
	- Updated Bundlesize GitHub Action to improve the reliability of bundle size checks.
	- Expanded node version support in linting workflow to include the latest version.
- **Enhancements**
	- Streamlined naming for linting workflow for better clarity.
- **Documentation**
	- Adjusted workflow documentation to reflect updated action versions and configurations.
- **Dependencies**
	- Added a new dependency (`minimist`) to improve command line parsing capabilities.

### CODE REVIEW

**Comment by gabrieljablonski** [comment]
The gh-actions commit (9319c87 for instance) seems to be "hiding" the last checks that were run.

image

Expected to see the checks, like this.

image

**Review by gabrieljablonski — APPROVED** [review]
Approving to see if checks are re-run. If not, changes should be made,

**Review by gabrieljablonski — CHANGES_REQUESTED** [review]
Nope 🤔

**Comment by danielbarion** [comment]
that's weird, I didn't realize it before, thanks for letting me know, do you have any idea on how to fix it?

File: .github/workflows/beta-release.yaml
Comment (gabrieljablonski) [inline .github/workflows/beta-release.yaml:31]: I believe "Auto Release Bot" is not recognized as an actual user, so it blocks the checks. Please try omitting this step, or setting to the committer to "github-actions[bot]".

Comment (danielbarion) [reply .github/workflows/beta-release.yaml:31]: this step is needed to save the current beta version into the package (to commit the changes), I can also set it to my user if the suggestion doesn't work, let's try

**Comment by danielbarion** [comment]
https://github.com/orgs/community/discussions/55906

**Comment by danielbarion** [comment]
Beta version released with the last commit 🚀

  ```
  yarn add react-tooltip@v5.25.2-beta.1148.5
  ```
  or
  ```
  npm install react-tooltip@v5.25.2-beta.1148.5
  ```

**Review by gabrieljablonski — APPROVED** [review]

**Pull request merged**
